client/src/app/app.component.ts

```diff
@@ -99,10 +105,14 @@ export class AppComponent implements OnInit {
     this.projectService
       .getAllProjects()
       .pipe(
-        catchError(() => {
+        catchError((response: HttpErrorResponse) => {
+          if (response.error.status === 401) {
+            // interceptor takes control and redirects to login
+            return EMPTY;
+          }
+          // show generic error page
           this.isLoading = false;
           this.isError = true;
-          return EMPTY;
```

Comment:
Does it work without returning (another) observable?

Reply:
No, returning an `EMPTY` observable is necessary ;)